src/algorithms/binary_histogram_threshold.jl

```diff
@@ -0,0 +1,36 @@
+struct BinaryHistogramThreshold{T} <: AbstractImageBinarizationAlgorithm
```

Comment:
`BinaryHistogramThreshold` seems not a good name, any suggestions?

Reply:
How about `ImageHistogramThreshold` ?

Reply:
The "Binary" here indicates that it's a method that uses only one threshold value.

Reply:
What about 
```julia
struct SingleHistogramThreshold{T} <: AbstractImageBinarizationAlgorithm
```